src/plugins/spark/components/widget/AnalogArrayEditor.vue

```diff
@@ -5,30 +5,32 @@ import {
   Block,
   BlockOrIntfType,
 } from 'brewblox-proto/ts';
-import { defineProps } from 'vue';
+import { computed } from 'vue';
 import { ENUM_LABELS_ANALOG_SENSOR_TYPE } from '@/plugins/spark/const';
 import { useSparkStore } from '@/plugins/spark/store';
+import { BlockAddress } from '@/plugins/spark/types';
 import { createBlockDialog } from '@/utils/block-dialog';
 import { createDialog } from '@/utils/dialog';
 
 interface Props {
   channels: AnalogModuleChannel[];
-  serviceId: string;
-  blockId: string;
+  address: BlockAddress;
```

Comment:
This will also need a call-side change from serviceId/blockId args to block address